### PR TITLE
Add a condition to include or exclude the processing of nitrate depending on the model's output

### LINKED ISSUE #765: Exclude nitrate in calculating Total AOD for UFS-Aerosols model (option: nasa_on) and update control files
1) The UFS-Aerosols model has remove the tracers of nitrate and ammonium. So the UPP also need to update the total AOD by removing the nitrate AOD.

2) Adding the control xml files and txt files for aerosols for GEFSv13

### PR DESCRIPTION
**Pull request by zhanglikate**
1. Add a condition to include or exclude the processing of nitrate depending on the model's output
2. Add  XML control files and TXT files for UFS-Aerosols model output.

### CODE REVIEW

File: parm/postcntrl_gefs_aerosol.xml [hunk 1]
```diff
@@ -10,64 +10,997 @@
     <sigreftime>fcst</sigreftime>
     <prod_status>oper</prod_status>
     <data_type>fcst</data_type>
-    <gen_proc_type>fcst</gen_proc_type>
+    <gen_proc_type>ens_fcst</gen_proc_type>
```
Comment (WenMeng-NOAA) [inline parm/postcntrl_gefs_aerosol.xml:13]: @zhanglikate The "ens_fcst" is not an official type of generating process at 
https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-3.shtml
Please revert it.

File: parm/postcntrl_gefs_aerosol.xml [hunk 2]
```diff
@@ -10,64 +10,997 @@
...
     <time_range_unit>hour</time_range_unit>
     <orig_center>nws_ncep</orig_center>
-    <gen_proc>gfs_avn</gen_proc>
+    <gen_proc>gefs</gen_proc>
```
Comment (WenMeng-NOAA) [inline parm/postcntrl_gefs_aerosol.xml:16]: @zhanglikate The "gefs" can't be identified as forecast generating process at
https://www.nco.ncep.noaa.gov/pmb/docs/on388/tablea.html
Please revert it.

**Comment by zhanglikate** [comment]
May I know how to revert it? Thanks.

Kate

File: parm/postcntrl_gefs_aerosol.xml (hunk 2 above)
Comment (zhanglikate) [reply parm/postcntrl_gefs_aerosol.xml:16]: May I know which GEFS control file that I need to following to add the aerosols output for GEFSv13? Thanks.

I am following postcntrl_gefs.xml to add the aerosols output?

**Comment by WenMeng-NOAA** [comment]
@zhanglikate You may add new aerosol fields in  the files postcntrl_gefs_aerosol*. I am meaning:
in postcntrl_gefs_aerosol.xml:
   Change
       <gen_proc_type>ens_fcst</gen_proc_type>
   into
        <gen_proc_type>fcst</gen_proc_type>

    change
         <gen_proc>gefs</gen_proc>
     into 
          <gen_proc>gfs_avn</gen_proc>

And the same changes for postcntrl_gefs_aerosol_f00.xml

File: parm/postcntrl_gefs_aerosol.xml [hunk 3]
```diff
@@ -13,61 +13,994 @@
     <gen_proc_type>fcst</gen_proc_type>
     <time_range_unit>hour</time_range_unit>
     <orig_center>nws_ncep</orig_center>
-    <gen_proc>gfs_avn</gen_proc>
+    <gen_proc>gfs</gen_proc>
```
Comment (WenMeng-NOAA) [inline parm/postcntrl_gefs_aerosol.xml]: @zhanglikate Please revert this change.

**Comment by zhanglikate** [comment]
I just updated and committed. Can you check?

File: parm/postcntrl_gefs_aerosol_f00.xml [hunk 4]
```diff
@@ -0,0 +1,1004 @@
...
+    <sub_center>ncep_nco</sub_center>
+    <version_no>v2003</version_no>
+    <local_table_vers_no>local_tab_yes1</local_table_vers_no>
+    <sigreftime>fcst</sigreftime>
+    <prod_status>oper</prod_status>
+    <data_type>fcst</data_type>
+    <gen_proc_type>fcst</gen_proc_type>
+    <time_range_unit>hour</time_range_unit>
+    <orig_center>nws_ncep</orig_center>
+    <gen_proc>gfs</gen_proc>
```
Comment (WenMeng-NOAA) [inline parm/postcntrl_gefs_aerosol_f00.xml]: @zhanglikate Please change "gfs" into "gfs_avn".

**Comment by zhanglikate** [comment]
Corrected.

**Comment by WenMeng-NOAA** [comment]
@zhanglikate and @lipan-NOAA Do you have UFS model output  for the UPP standalone test? The files either on WCOSS2 or Hera would work for me.

**Comment by zhanglikate** [comment]
> @zhanglikate and @lipan-NOAA Do you have UFS model output for the UPP standalone test? The files either on WCOSS2 or Hera would work for me.

@WenMeng-NOAA This the model results at HERA: /scratch1/BMC/gsd-fv3-dev/lzhang/AODtest/case2

**Comment by WenMeng-NOAA** [comment]
@zhanglikate To generate GEFSV13 aerosol products in UPP, which option should I use in my test, gccpp_on, nasa_on, rdaod?

**Comment by zhanglikate** [comment]
nasa_on. Thanks

**Comment by lipan-NOAA** [comment]
example file => HPSS: /NCEPDEV/emc-naqfc/1year/Li.Pan/tmp/forkate-atmnc-upptest-20180808.htar & forkate-sfcnc-upptest-20180808.htar

**Comment by WenMeng-NOAA** [comment]
@zhanglikate You may add the following changes in parm/postcntrl_gefs_aerosol.xml and parm/postcntrl_gefs_aerosol_f00.xml to label ensemble products:
```
change
    <gen_proc_type>fcst</gen_proc_type>
into
   <gen_proc_type>ens_fcst</gen_proc_type>
change
    <gen_proc>gfs_avn</gen_proc>
into
   <gen_proc>gefs</gen_proc>
    
```
After update, the relevant txt files need to regenerated.

**Comment by zhanglikate** [comment]
> ens_fcst

@WenMeng-NOAA  Finished. Can you check? Thanks.

**Comment by WenMeng-NOAA** [comment]
> > ens_fcst
> 
> @WenMeng-NOAA Finished. Can you check? Thanks.

@zhanglikate Your PR look good for me. I will run testing now. Thanks!

**Comment by WenMeng-NOAA** [comment]
@zhanglikate and @lipan-NOAA Do you know if the file parm/AEROSOL_LUTS.dat is used for processing UFS based aerosol data? If not, can we delete it?

**Comment by zhanglikate** [comment]
These are for GEFS-Aerosols, better to keep them. Thanks.

Kate

**Comment by WenMeng-NOAA** [comment]
@zhanglikate My test is at /scratch1/NCEPDEV/stmp2/Wen.Meng/post_gefs_2018080800 on Hera. Please let me know if you see any discrepancies with your test.

**Review by lipan-NOAA — APPROVED** [review]

**Comment by lipan-NOAA** [comment]
I tested it on WCOSS2, it looks fine for me

**Comment by WenMeng-NOAA** [comment]
> I tested it on WCOSS2, it looks fine for me

@lipan-NOAA Can you provide me the model output files in netcdf from your test on WCOSS2?

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA You may start the UPP RT on Hera and Orion. There should be updates in job scripts of gefs testing with aerosol fix file relocation. You may pick up my changes from
```
Hera: /scratch2/NCEPDEV/ovp/Wen.Meng/ncep_post/post_regression_test_new/jobs
         run_post_fv3gefs_HERA.sh and run_post_fv3gefs_pe_test_HERA.sh

Orion: /home/wmeng/ovp/ncep_post/post_regression_test_new/jobs
         run_post_fv3gefs_ORION.sh and run_post_fv3gefs_pe_test_ORION.sh
```

**Comment by WenMeng-NOAA** [comment]
The UPP RT was completed on WCOSS2 without changed results.

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA You may start the UPP RT on Hera and Orion. There should be updates in job scripts of gefs testing with aerosol fix file relocation. You may pick up my changes from
> 
> ```
> Hera: /scratch2/NCEPDEV/ovp/Wen.Meng/ncep_post/post_regression_test_new/jobs
>          run_post_fv3gefs_HERA.sh and run_post_fv3gefs_pe_test_HERA.sh
> 
> Orion: /home/wmeng/ovp/ncep_post/post_regression_test_new/jobs
>          run_post_fv3gefs_ORION.sh and run_post_fv3gefs_pe_test_ORION.sh
> ```

Apologies for the delay, I have RTs running on Hera and Orion. They've been temperamental today and dropping connection / freezing a lot from my side. I'll post results when they finish.

**Comment by FernandoAndrade-NOAA** [comment]
UPP RTs have completed successfully on Hera and Orion with no changes to results.

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.
